### ISSUE TITLE
Burnout 3/Revenge/Black - Missing sky texture

### ISSUE THREAD

**Issue by karasuhebi** [PCSX2/pcsx2#563]
-PCSX2 version: 1.3.1-382

-PCSX2 options: Defaults.

-Plugins used: GSdx32-AVX2, SPU2-X, LilyPad, cdvdGigaherz, the rest of the plugins were null.

-Plugin settings: Tested GSdx D3D9, D3D11 and OGL, both hardware and software renderers. Used both native and upscaled resolutions. Also tried with AF off and AF 16x.

-Description: The game's sky box appears black. Switching to software mode won't fix the problem. The issue is fixed if you boot into the game with a software renderer.

-How to replicate: Boot the game and play it. The issue shows up in any race.

-Last known version to work: Unknown, probably never worked.

-PC specifications: CPU: Intel Core i5-4670K @ 4.10GHz GPU: Sapphire HD 7950 w/Boost 3GB OS: Windows 8.1 Pro

-Other comments: If you boot into the game after already having played another game (in my case, DRIV3R), the missing sky box will be filled with garbage data, possibly from the previous game.

Screenshots:
http://imgur.com/a/0JhH9

Bonus screenshot of the blur effect in the game being broken (turned into ghosting):
http://i.imgur.com/sNTUBoY.png <- Look on the right side of the car.

**Comment by ssakash** [comment]
GS dump is needed to debug the graphical glitches,

**Comment by bositman** [comment]
The game has CRC hacks so you should also disable CRCs in GSdx to actually see all that is broken (and it's a lot :P)

**Comment by karasuhebi** [comment]
Ah yes, GS dump is probably a good idea. Thank you @ssakash. @bositman brings a good point though. @gregory38, would you like a GS dump with all CRC hacks disabled?

**Comment by gregory38** [comment]
Just do the dump in sw mode.

**Comment by karasuhebi** [comment]
https://dl.dropboxusercontent.com/u/6425650/gsdx_20150529031828.bmp
https://dl.dropboxusercontent.com/u/6425650/gsdx_20150529031828.gs

**Comment by gregory38** [comment]
Did you try with 
```
UserHacks = 1
UserHacks_DisableCrcHacks = 1
UserHacks_AutoSkipDrawDepth = 0
```

Lots of effect are disabled. And depth textures aren't read back. I can't reproduce the issue but I think it would be better to fix all effects with known issue first.

The sky issue could be related to 16 bits / 32 bits conversion. A 32 bits 640x480 target is equivalent to a 16 bits 640x960 target but guess what, gsdx still uses 640x480, and half of the screen is potentially garbage...

**Comment by karasuhebi** [comment]
I did not try with any hardware hacks, no. I will do so now.

**Comment by gregory38** [comment]
And try also to disable the texture cache (if you know how to recompile gsdx).
Config.h, => use this line
```
#define DISABLE_HW_TEXTURE_CACHE // Slow but fixes a lot of bugs
```

**Comment by karasuhebi** [comment]
OMG, without CRC hacks:
http://i.imgur.com/FQZLj2O.png

**Comment by gregory38** [comment]
yes I said various effect aren't emulated correctly ;)

**Comment by karasuhebi** [comment]
I'll disable texture cache in a minute and get back to you.

**Comment by karasuhebi** [comment]
Yeah the sky is still missing with hardware texture cache disabled. Want a GS dump?

**Comment by ssakash** [comment]
what about ZeroGS plugin ? was the sky still present ?

**Comment by karasuhebi** [comment]
ZeroGS: http://i.imgur.com/jqTGLX4.png

FWIW, ZeroGS doesn't even let me see the menus. It shows me the Loading progress bar when the game first boots and then you see NOTHING until you get into a race. Then you get what you see above.

**Comment by karasuhebi** [comment]
Would someone mind booting some random game and then booting into Burnout 3 using full boot instead of fast to see if the garbage textures are still there or if they get flushed??

Thanks!

**Comment by yorkiesandskittles** [comment]
I full booted into Burnout 1, played a race and then full booted into Burnout 3. Then the game was suddenly on acid:
http://i.imgur.com/6xNQ08u.jpg

It seems like the textures ARE THERE, but the coloring is wrong. However, notice how the far right of the sky is just grey.

**Comment by karasuhebi** [comment]
Doubt D3D11 will be any different. That's the same issue I was having. I just wanted to see if doing a full boot did anything to flush the garbage textures that stick around from the previous game.

Thanks!

**Comment by karasuhebi** [comment]
@gregory38 Should I test this game now with the latest build?

**Comment by ssakash** [comment]
I think it'd be nice to check whether bug is present with Preload data frame option.

**Comment by yorkiesandskittles** [comment]
I just tested it now. This is what happened when I full-booted into Burnout 1, played a race, **reset to Burnout 3** and played a Time Attack race on SLS, with HW mode the whole time. 

There doesn't seem to be a difference with Preload Data Frame on.

![b3_sky_latest](https://cloud.githubusercontent.com/assets/8225831/11343807/aea80d68-9205-11e5-8d5b-72de8e1f4fcb.png)

Switching to SW and back to HW removes the pink/purple layer from the screen, but the skybox remains as is:

![b3_sky_latest_sw_to_hw](https://cloud.githubusercontent.com/assets/8225831/11343843/f5aa6ba2-9205-11e5-9879-8ddfc2ff1bc4.png)

Also, shouldn't it be Preload **Frame Data**?

**Comment by yorkiesandskittles** [comment]
For some reason, using 50Hz produces different results to 60Hz on the PAL version. When you enter the garage on 60Hz, the car's windows will be a strange colour (green in my testing), but on 50Hz, this issue is not present at all with the same settings. The sky is also less glitched on 50Hz.

**Comment by joder666** [comment]
@ssakash Yes, sadly it is still present. In the other hand, unrelated to this, i believe the missing blur effect when boosting is fixed.

**Comment by FlatOutPS2** [comment]
> The issue is fixed if you boot into the game with a software renderer.
You don't need to boot the game with software renderer, just loading a track in software mode is enough. If you've loaded a track with a hardware renderer and get black skies, then switch to software mode and exit back to the menu in software mode, then switch back to the hardware renderer and use that to load a track, the sky textures will also be there, but the sky will be white(the blue sky color won't get loaded).

**Comment by marios237** [comment]
![image](https://user-images.githubusercontent.com/34271773/33605120-0a9d51fa-d9c1-11e7-9176-7e8a3cbe82ea.png)
what is that can anyone help me ???

**Comment by MrCK1** [comment]
@marios237 This isn't a place for general support. If you have a question, please make a thread at [forums.pcsx2.net](url) instead.

**Comment by tadanokojin** [comment]
Issue is as follows (for Revenge):

Game is rendering sky texture (note *rendering*, no EE write) to buffer at `0x2c00`. However, actual sky data is offset +32 pixels in the x direction. This happens once as the game is in the loading screen.

Later, game will sample `0x2c10` for sky data with no offset. Since the cache is not designed to deal with this situation, it results in a miss. This issue is similar to the notorious Jak Eye issue. Texture inside rt will alleviate this but it will also result in some bugs dealing with reflections on the cars.

I'll work with @iMineLink for a solution.

**Comment by lightningterror** [comment]
Fixed in #4094
